### PR TITLE
Check if title or description is null

### LINKED ISSUE #18: iOS crash, empty IAPProduct
Sometimes, when developer create new purchase, AppStore returns IAPProduct with empty fields (except productIdentifier) and app crash on the following line: 
https://github.com/defold/extension-iap/blob/8d9ea79d7ddef8bc94035f510907008db938c233/extension-iap/src/iap_ios.mm#L122

I think extension should check if this field empty and provide debug message. 

I found that after a few renaming of purchases (yes, just rename purchase in App Store Connect, and then rename it back), I start receiving valid purchase info.

More info here: https://github.com/defold/editor2-issues/issues/2908

### PR DESCRIPTION
**Pull request by britzl**
Fixes #18

### CODE REVIEW

File: extension-iap/src/iap_ios.mm [hunk 1]
```diff
@@ -119,8 +119,8 @@ - (void)productsRequest:(SKProductsRequest *)request didReceiveResponse:(SKProdu
 
         IAPProduct product = {0};
         product.ident           = strdup([p.productIdentifier UTF8String]);
-        product.title           = strdup([p.localizedTitle UTF8String]);
-        product.description     = strdup([p.localizedDescription UTF8String]);
+        product.title           = p.localizedTitle ? strdup([p.localizedTitle UTF8String]) : strdup([p.productIdentifier UTF8String]);
```
Comment (britzl) [inline extension-iap/src/iap_ios.mm]: Should we return an empty string or the product id if title doesn't exist?

**Comment by britzl** [comment]
@AGulev From the looks of the SKProduct there was a price and local price string right? It seemed like it was only the localized title and description that was null.

File: extension-iap/src/iap_ios.mm (hunk 1 above)
Comment (AGulev) [reply extension-iap/src/iap_ios.mm]: I think, yes. Maybe it makes sense to show warning for a developer?

**Comment by AGulev** [comment]
it's hard to say now. I can't even reproduce this issue anymore (it seems like it was app store bug), that's why I think it makes sense to show a warning for the developer if it's nil.
In this case, even if we miss something it will be much easier to understand the reason of the crash

**Comment by britzl** [comment]
Ok, so you're saying warning and then crash instead of warning and trying to recover? If we're sure it's a problem that the developer can handle then sure, a crash is better.

**Comment by AGulev** [comment]
I think warning and trying to recover is fine in this case. If developer find that he doesn't have title he starts digging and find warning that we don't get title from the store.

**Pull request merged**